tests/unit/test_artifactory_path.py

```diff
@@ -910,20 +935,38 @@ def test_deploy_file(self):
             file.write("I am a test file")
 
             constructed_url = f"{path}{test_file.name};{matrix_parameters}"
-            responses.add(
-                responses.PUT, constructed_url, status=200, match_querystring=True
-            )
-
-            path.deploy_file(
-                test_file,
-                explode_archive=True,
-                explode_archive_atomic=True,
-                parameters={
-                    "deb.architecture": "amd64",
-                    "deb.component": "contrib",
-                    "deb.distribution": ["dist1", "dist2"],
-                },
-            )
+            responses.add(responses.PUT, constructed_url, status=200)
+
+            # can't use pytest.mark.parametrize with unittest classes
+            for quote_params in (None, True, False):
+                with mock.patch(
+                    "urllib.parse.quote", new=mock.Mock(wraps=quote_original)
+                ) as q, mock.patch("artifactory.warn") as w:
+                    path.deploy_file(
+                        test_file,
+                        explode_archive=True,
+                        explode_archive_atomic=True,
+                        parameters={
+                            "deb.architecture": "amd64",
+                            "deb.component": "contrib",
+                            "deb.distribution": ["dist1", "dist2"],
+                        },
+                        quote_parameters=quote_params,
+                    )
+                    # TODO: v0.10.0 - None test will not be needed, warn test will not be needed
+                    if quote_params is None:
+                        w.assert_called_once_with(
```

Comment:
There's built-int tools for asserting warning, I think
- UnitTest - https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarnsRegex
- Pytest - https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html

I don't remember which one we run on CI, but let's use it's tool!

Reply:
It seems we execute with pytest, but many/most of the tests are written with UnitTest classes/assertions (this is common), and only a few are written in actual pytest style; I think I saw some comments about wanting to convert them all eventually, which would be very welcome indeed as I find pytest much nicer personally.

Anyway I will look to replace this with some form of built-in warning assertion, thanks for pointing this out (this code will also be gone in v0.10.0 either way though).

Reply:
I have updated this in https://github.com/devopshq/artifactory/pull/409/commits/65b2889d5791da174fb72d010717090dc3d40275 , however in doing so:
* the warnings are no longer suppressed so they show in the test output
* there doesn't seem to be a way to assert that no warnings were issued with this method (UnitTest)
  * this is a UnitTest class so I think I cannot use the pytest methods that involve needing to add a fixture call (`recwarn`)
  * using the context manager instead to check for no warnings would probably work, but I will have to duplicate the call to `deploy` then depending on whether warnings should be raised or not

There may be ways to work around the above that I'm not seeing, but since this only needs to exist for one version, I'm not too concerned about regressions anyway. Let me know if this works for you!

Reply:
@allburov could you evaluate the above please? I believe this satisfies your ask.